File: backend/Application/Mappers/ToolingMapper.cs
using Application.DTOs.Responses;
using Application.Interfaces;
using Domain.Entities;

namespace Application.Mappers;

public class ToolingMapper : IToolingMapper
{
    public ToolingResponse Map(Tooling tooling)
    {
        return new ToolingResponse
        {
            Id = tooling.Id,
            Name = tooling.Name,

            Media = tooling.Media is null
                ? null
                : new MediaResponse
                {
                    Path = tooling.Media.Path,
                    AltText = tooling.Media.AltText,
                    Type = tooling.Media.Type
                }
        };
    }
}

File: frontend/src/mappers/toolingMapper.js
export function mapTooling(data) {
    if (!data) {
        throw new Error("Tooling data is missing");
    }

    return {
        id: data.id,
        name: data.name ?? "",
        media: data.media
            ? {
                path: data.media.path ?? "",
                altText: data.media.altText ?? "",
                type: data.media.type,
            }
            : null,
    };
}